bin/reth/src/args/network_args.rs

```diff
@@ -54,26 +54,26 @@ impl NetworkArgs {
         &self,
         config: &Config,
         chain_spec: Arc<ChainSpec>,
-    ) -> NetworkConfigBuilder {
-        let peers_file = (!self.no_persist_peers).then_some(&self.peers_file);
+    ) -> eyre::Result<NetworkConfigBuilder> {
         let network_config_builder = config
-            .network_config(self.nat, peers_file.map(|f| f.as_ref().to_path_buf()))
+            .network_config(self.nat, self.persistent_peers_file()?)
             .boot_nodes(self.bootnodes.clone().unwrap_or_else(mainnet_nodes))
             .chain_spec(chain_spec);
 
-        self.discovery.apply_to_builder(network_config_builder)
+        Ok(self.discovery.apply_to_builder(network_config_builder))
     }
 }
 
 // === impl NetworkArgs ===
 
 impl NetworkArgs {
     /// If `no_persist_peers` is true then this returns the path to the persistent peers file
-    pub fn persistent_peers_file(&self) -> Option<PathBuf> {
+    pub fn persistent_peers_file(&self) -> eyre::Result<Option<PathBuf>> {
         if self.no_persist_peers {
-            return None
+            return Ok(None)
         }
-        Some(self.peers_file.clone().into())
+        self.peers_file.as_ref().parent().map(std::fs::create_dir_all).transpose()?;
+        Ok(Some(self.peers_file.clone().into()))
```

Comment:
hmm, I'm not sure we should create this here. because this function is supposed to return only the path.

this should be handled when the file is written.

Reply:
i encountered this error, while it attempted to read the file, but agreed. should be handled differently